src/main/java/mods/railcraft/common/carts/CartBaseMaintenance.java

```diff
@@ -12,32 +12,115 @@
+import mods.railcraft.common.gui.buttons.StandardButtonTextureSets;
+import mods.railcraft.common.gui.tooltips.ToolTip;
 import mods.railcraft.common.plugins.forge.DataManagerPlugin;
+import mods.railcraft.common.plugins.forge.LocalizationPlugin;
 import mods.railcraft.common.plugins.forge.WorldPlugin;
 import mods.railcraft.common.util.inventory.InvTools;
 import mods.railcraft.common.util.misc.Game;
+import mods.railcraft.common.util.network.IGuiReturnHandler;
+import mods.railcraft.common.util.network.RailcraftInputStream;
+import mods.railcraft.common.util.network.RailcraftOutputStream;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockRailBase;
 import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.network.datasync.DataParameter;
 import net.minecraft.network.datasync.DataSerializers;
+import net.minecraft.util.IStringSerializable;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
+import org.jetbrains.annotations.Nullable;
 
+import java.io.IOException;
 import java.util.List;
+import java.util.Locale;
 
 /**
  * @author CovertJaguar <http://www.railcraft.info>
  */
-public abstract class CartBaseMaintenance extends CartBaseContainer {
+public abstract class CartBaseMaintenance extends CartBaseContainer implements IGuiReturnHandler {
 
     private static final DataParameter<Byte> BLINK = DataManagerPlugin.create(DataSerializers.BYTE);
     protected static final double DRAG_FACTOR = 0.9;
-    protected static final float MAX_SPEED = 0.1f;
     private static final int BLINK_DURATION = 3;
     private static final int DATA_ID_BLINK = 25;
+    public static final DataParameter<Byte> CART_MODE = DataManagerPlugin.create(DataSerializers.BYTE);
+    private final MultiButtonController<CartModeButtonState> modeController = MultiButtonController.create(0, CartModeButtonState.VALUES);
+    public CartMode clientMode = CartMode.SERVICE;
+
+    @Override
+    public float getMaxCartSpeedOnRail() {
+        return getMode().speed;
+    }
+
+    public enum CartMode implements IStringSerializable {
+
+        SERVICE(0.1f),
+        TRANSPORT(0.4f);
+        public static final EntityCartTrackLayer.CartMode[] VALUES = values();
+        public final float speed;
+
+        CartMode(float speed) {
+            this.speed = speed;
+        }
+
+        @Override
+        public String getName() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    public MultiButtonController<CartModeButtonState> getModeController() {
+        return modeController;
+    }
+
+    public enum CartModeButtonState implements IMultiButtonState {
+        SERVICE(LocalizationPlugin.translate("gui.railcraft.cart.maintenance.mode.service")),
+        TRANSPORT(LocalizationPlugin.translate("gui.railcraft.cart.maintenance.mode.transport"));
```

Comment:
Shouldn't translate immediately but only on demand; otherwise the translation will not update once you update language on client

Reply:
Done by making `getLabel` return the localized button label.